azerothcore/azerothcore-wotlk
# feat(Core/Loot): Implement player_loot_template

* Also fix AV player loot



## Changes Proposed:
-  Implement player_loot_template table (entry = Team ID)
-  This is a cleaner approch than just hacking a creature_loot_template and sending it and should also allow modules to do cool stuff.
- Implement CONDITION_SOURCE_TYPE_PLAYER_LOOT_TEMPLATE (28), same as all the other loot conditions

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/9133

## SOURCE:


## Tests Performed:

- Tested in-game.
- 


## How to Test the Changes:


1. Join Alterac Valley & check if removing insignia gives you items
2. Join another BG and check if removing insignia gives you items
3.

## Known Issues and TODO List:


- [ ] Wintergrasp quest item loot (currently broken) will still need to be fixed (outside the scope of this PR, but should be simple enough)


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
